src/bfloat16.jl

```diff
@@ -424,3 +424,6 @@ for F in (:abs, :abs2, :sqrt, :cbrt,
      Base.$F(x::BFloat16) = BFloat16($F(Float32(x)))
   end
 end
+
+# irrationals
+BFloat16(num::Irrational) = BFloat16(Float32(num))
```

Comment:
```suggestion
BFloat16(x::AbstractIrrational) = BFloat16(Float32(x)::Float32)
```

`num` -> `x` for consistency with the other conversion/constructors, `AbstractIrrational` instead of `Irrational` since that's how the other euqivalent `Base` functions are defined, and the type assert to ensure the return type is properly inferred.

Ex before:
```
julia> using BFloat16s; Base.return_types(BFloat16, [Irrational])
1-element Vector{Any}:
 Any
```

Ex. after:
```
julia> using BFloat16s; Base.return_types(BFloat16, [AbstractIrrational])
1-element Vector{Any}:
 BFloat16
```